### PR TITLE
Onboard Flow onto validation components [SCI-1725]

### PR DESCRIPTION
**Pull request by Ducz0r**
This PR adds Flow onto validation components. They're not 100% covered, but mostly vary in the 70-90% range.

I think that for certain things (like context, and certain React shenanigans), Flow is still very immature and cumbersome to use. Also it sucks that you cannot make a reusable function definition (that I could reuse for validators), you can only create class interfaces, which I think is a bit of an overkill right now).

Closes [SCI-1725](https://biosistemika.atlassian.net/browse/SCI-1725).

### CODE REVIEW

File: app/javascript/src/components/validation/components/ValidatedSubmitButton.jsx [hunk 1]
```diff
@@ -1,17 +1,19 @@
-import React from "react";
+// @flow
+
+import * as React from "react";
 import { Button } from "react-bootstrap";
 import PropTypes from "prop-types";
 
-const ValidatedSubmitButton = (props, context) =>
+type Props = {
+  children?: React.Node
```
Comment (ZmagoD) [inline app/javascript/src/components/validation/components/ValidatedSubmitButton.jsx]: you can just import { Node } form react here we don't need all React types

**Comment by ZmagoD** [comment]
Also run flow-typed install to update the types :banana:

**Comment by Ducz0r** [comment]
@ZmagoD just double-checking if I imported the updated `flow-types` correctly? So I don't merge something bad 🙂.

File: package.json [hunk 2]
```diff
@@ -54,6 +53,8 @@
     "css-loader": "^0.28.4",
     "extract-text-webpack-plugin": "^3.0.0",
     "file-loader": "^0.11.2",
+    "flow-bin": "^0.56.0",
```
Comment (ZmagoD) [inline package.json]: Why you moved flow-bin and flow-typed from dev dependencies?

File: .eslintrc.json [hunk 3]
```diff
@@ -19,6 +19,8 @@
   "rules": {
     "import/extensions": "off",
     "import/no-unresolved": "off",
+    // Because of flow-typed & flow-bin
+    "import/no-extraneous-dependencies": ["error", { "devDependencies": true }],
```
Comment (ZmagoD) [inline .eslintrc.json]: :+1:

File: package.json (hunk 2 above)
Comment (Ducz0r) [reply package.json]: Because ESLint was telling me so :banana:. I've updated the configuration.

**Pull request merged**